### PR TITLE
Add OOS skus for configurable product price calculation

### PR DESCRIPTION
**Pull request by dairbuirabass**
## Description
At the moment, module sets configurable product price with OOS skus to be 0, since the builder only checks the price of the inStock skus. These changes remove this functionality and check through all sku prices whether they are in or out of stock.

For this to work merchant should enable the `Display Out of Stock Products` setting to `Yes` because setting it to `No` will remove the prices from `catalog_product_index_price` table which Nosto_Tagging module relies on when building price data for products.

## Related Issue
-

## Motivation and Context
Product price should not be set to 0 even when product is OOS

## How Has This Been Tested?
Locally M2.4

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Model/Product/Repository.php [hunk 1]
```diff
@@ -312,11 +305,7 @@ private function saveParentIdsToCache(ProductInterface $product, $parentProductI
      */
     public function getSkusAsArray(Product $product, Store $store)
     {
-        $inStockProductsByIds = $this->stockProvider->getInStockProductIds(
```
Comment (olsi-qose) [inline Model/Product/Repository.php:315]: This will introduce another bug where the product price would be incorrect if the cheapest sku is out-of-stock

**Review by olsi-qose — CHANGES_REQUESTED** [review]

**Comment by olsi-qose** [comment]
If all skus are out-of-stock then the configurable product will be out-of-stock too.

File: Model/Product/Repository.php (hunk 1 above)
Comment (dairbuirabass) [reply Model/Product/Repository.php:315]: @olsi-qose ah, so the "as low as" price is dynamically changing in Nosto depending on the stock status?

**Comment by dairbuirabass** [comment]
>

**Pull request closed without merging**